Duplicati/Library/Main/Database/LocalDatabase.cs

```diff
@@ -1283,8 +1314,8 @@ public virtual long CreateFileset(long volumeid, DateTime timestamp, System.Data
             using (var cmd = m_connection.CreateCommand())
             using (var tr = new TemporaryTransactionWrapper(m_connection, transaction))
             {
-                cmd.Transaction = tr.Parent;                
-                var id = cmd.ExecuteScalarInt64(@"INSERT INTO ""Fileset"" (""OperationID"", ""Timestamp"", ""VolumeID"") VALUES (?, ?, ?); SELECT last_insert_rowid();", -1, m_operationid, NormalizeDateTimeToEpochSeconds(timestamp), volumeid);
+                cmd.Transaction = tr.Parent;
+                var id = cmd.ExecuteScalarInt64(@"INSERT INTO ""Fileset"" (""OperationID"", ""IsFullBackup"", ""Timestamp"", ""VolumeID"") VALUES (?, ?, ?, ?); SELECT last_insert_rowid();", -1, m_operationid, BackupType.PARTIAL_BACKUP, NormalizeDateTimeToEpochSeconds(timestamp), volumeid);

```

Comment:
The `CreateFileset` method is also invoked by the `RecreateDatabaseHandler`.  Have we ensured that the `IsFullBackup` field is properly restored after a database recreate operation?

Reply:
I think it will currently default to partial. We can't know what the original state was since it isn't saved in the remote backup so it should default to full.

Reply:
That's true, we cannot recreate the original state since the backup type is not preserved.  

However, I think this does lead to some strange behavior that users might not anticipate.  For example:
1. Create a full backup.
2. Create partial backup.
3. Recreate database (now both filesets are marked as a full backups)
4. Retention policy (or something else) removes the first backup, leaving only the partial backup. 

Also, are you able to reproduce the following?

1. Create a partial backup.  The main page shows 1 backup version, and the restore list shows 1 (partial) version.
2. Create a full backup.  The main page shows 1 backup version, and the restore list shows 1 (full) version.
3. Recreate the database.  The main page now shows 1 backup version, but the restore list shows 2 versions.

Reply:
I've finally made it through all the changes.  I'll start to run some more tests to see if anything unexpected reveals itself.  Thanks again for your patience @BlueBlock.

I do have a few remaining questions:
1. Any thoughts about the two situations described above?
2. Do you think this should be included in the next beta?  I believe this will address some of the causes of database corruption, but my only concern is the amount of testing given the large number of changes.  Given that the next beta is likely not _that_ soon, one approach is to create more canary releases to get these changes out there for users to test.

Reply:
All those are a result of not having the state stored at the remote end. To avoid those scenarios it may be worthwhile to see what it would take to store the state though.. I'll have a look. Once that gets decided then I think we can include it in a canary then beta.

Reply:
I believe the number of versions does sync up once the next backup is run...perhaps this is a minor issue?